server/cloudformation/ecs.go

```diff
@@ -196,3 +212,16 @@ func requiresReplacement(new, old *ECSServiceProperties) bool {
 
 	return false
 }
+
+func getPrimaryDeployment(service *ecs.Service) (deployment *ecs.Deployment, err error) {
```

Comment:
You can just change this function to:

```go
func primaryDeployment(service *ecs.Service) (*ecs.Deployment) {
        for _, d := range service.Deployments {
                if d.Status != nil && *d.Status == "PRIMARY" {
                        return d
                }
        }
        return nil
}
```

And use the nil value to indicate that there's no primary deployment, then return the error elsewhere.